### PR TITLE
Refactor collision detection

### PR DESCRIPTION
**Pull request by brianrip**
@damwhit @cheljoh @rrgayhart 

This moves collision logic into the copter class. By doing this we are able to extract jQuery calls and test collision detection at  the unit level.

Look at the original game.js file and checkForCollision function. Then follow the changes from there as the function is removed and places in the copter.js. 

Clone the repo - npm install - npm run build - npm start - visit: localhost:8080 - Play game and fly copter into obstacle or boundary.

I felt it was important to have the collision detection tested and jQuery calls were preventing that.

### CODE REVIEW

File: test/copter-test.js [hunk 1]
```diff
@@ -48,4 +48,17 @@ describe('Copter', function() {
...
+      var img = 'img';
+      var ycoord = 495;
+      var copter = new Copter(img, 0, ycoord, 10, 10);
+      var boundary = new Boundary(0, 500, 200, 10);
+      assert.equal(copter.status, "flying");
+      copter.gravity();
+      copter.checkForCollision(boundary);
+      assert.equal(copter.status, "crashed");
+    });
+  });
```
Comment (damwhit) [inline test/copter-test.js:63]: I like it!  Love those juicy tests.  I'm going to merge to master and 🛳 it.

File: lib/copter.js [hunk 2]
```diff
@@ -24,9 +23,19 @@ class Copter {
   this.downpull = - 6;
   }
 
-  crash() {
-    $('.game-canvas').hide();
-    $('.end-game').show();
+  checkForCollision(collider) {
+    if (this.x < collider.x + collider.width &&
```
Comment (cheljoh) [inline lib/copter.js:27]: I would pull the contents of the if statement into another helper function. Would make it easier to test.

File: lib/game.js [hunk 3]
```diff
@@ -187,8 +171,9 @@ function isObstacle(collider){
   return collider instanceof Obstacle;
 }
 
-function isPowerUp(collider){
-  return collider instanceof PowerUp;
+function crash(){
+  $('.game-canvas').hide();
+  $('.end-game').show();
 }
```
Comment (cheljoh) [inline lib/game.js:178]: I like this refactoring!

File: lib/obstacle.js [hunk 4]
```diff
@@ -30,6 +32,10 @@ class Obstacle {
   shuffleObstacle() {
     this.y = Math.floor((Math.random() * (450 - 80)) + 80);
   }
+
+  isPowerUp(collider) {
+    return collider instanceof PowerUp;
+  }
 }
```
Comment (cheljoh) [inline lib/obstacle.js:40]: 🍡

File: test/copter-test.js (hunk 1 above)
Comment (cheljoh) [inline test/copter-test.js:63]: Yeah nice testing!

File: lib/game.js (hunk 3 above)
Comment (damwhit) [inline lib/game.js:178]: I think that it was a good Idea to pull this out into a private method since it does not need to be called on a specific object.

**Comment by cheljoh** [comment]
I think this is good to merge! Nice jorb!

File: lib/copter.js (hunk 2 above)
Comment (rrgayhart) [inline lib/copter.js:27]: Upvote ^

File: test/copter-test.js (hunk 1 above)
Comment (rrgayhart) [inline test/copter-test.js:63]: Yay!

**Comment by rrgayhart** [comment]
Nice Work! Keep an eye on matching the requirements for assignments, even if they seem weird. For example, this PR didn't use the PR template as requested - not a big deal because you provide good detail, but ignoring that requirement would look really bad on a code challenge for a company.

**Comment by brianrip** [comment]
@rrgayhart I used the template but I removed the questions as I felt it redundant to state what I was going to state. Next time I will leave them : )

**Pull request merged**